pkg/integrations/manager.go

```diff
@@ -224,25 +227,45 @@ func (m *Manager) runIntegration(ctx context.Context, cfg Config, i Integration)
 	}
 }
 
-func (m *Manager) instanceConfigForIntegration(cfg Config, i Integration) instance.Config {
+func (m *Manager) instanceConfigForIntegration(cfg Config, serverConfig serverconfig.ServerConfig, i Integration) instance.Config {
 	prometheusName := fmt.Sprintf("integration/%s", cfg.Name())
 
 	common := cfg.CommonConfig()
 	relabelConfigs := append(m.defaultRelabelConfigs, common.RelabelConfigs...)
 
 	var scrapeConfigs []*config.ScrapeConfig
+	schema := "http"
+
+	// If there is a TLS cert path then assume we need to use client certificates
+	httpClientConfig := config_util.HTTPClientConfig{}
+	if serverConfig.HTTPTLSConfig.TLSCertPath != "" {
+		if serverConfig.ClientKey == "" || serverConfig.ClientCert == "" {
```

Comment:
Hmm, I'm not sure this will be quite right. Even if the HTTPS server has a TLS cert provided, we only need a client cert if the server is trying to validate client certs, which it shouldn't be doing by default.

Reply:
I need to test a few more scenarios. Will probably expound on this and make this is a check if client_auth_type == client_auth_type.